### ISSUE TITLE
Also check comprehensions with B007

### ISSUE THREAD

**Issue by Hoxbro** [astral-sh/ruff#9454]
[B007](https://docs.astral.sh/ruff/rules/unused-loop-control-variable/) (unused-loop-control-variable) right now only checks a normal for loops. I think this could be expanded so it also checks the different kinds of comprehension.

Example of code that should be safe to fix with `y` → `_y` (correct me if I'm wrong):
``` python
(x for x, y in zip(xs, ys))
[x for x, y in zip(xs, ys)]
{x for x, y in zip(xs, ys)}
{x: x for x, y in zip(xs, ys)}
{x: "value" for x, y in zip(xs, ys)}
{"key": x for x, y in zip(xs, ys)}
```

No unused-variables examples
``` python
(x for x, y in zip(xs, ys) if y == 1)
[x for x, y in zip(xs, ys) if y == 1]
{x for x, y in zip(xs, ys) if y == 1}
{x: x for x, y in zip(xs, ys) if y == 1}
(x if y == 1 else 0 for x, y in zip(xs, ys))
[x if y == 1 else 0 for x, y in zip(xs, ys)]
{x if y == 1 else 0 for x, y in zip(xs, ys)}
{x: x if y == 1 else 0 for x, y in zip(xs, ys)}
```

Edge case of an unused variable which cannot be renamed:
``` python
(_y for _y, y in zip(xs, ys) if _y == 1)
```

**Comment by zanieb** [comment]
In all those cases wouldn't we want to suggest _not_ zipping the variables together in the first place? If those are just trivial examples for iteration over some collection of tuples, I think I agree it'd make sense to suggest marking it as a `unused-loop-control-variable` although we may want a separate rule like `unused-comprehension-variable`?

**Comment by Hoxbro** [comment]
> If those are just trivial examples for iteration over some collection of tuples

They were meant as trivial examples. 

My example just did what B007 does for zip in a for loop ([playground](https://play.ruff.rs/7ef36098-459d-44f9-82f8-e63249be676f)). I agree; it would be much better to remove the zip in this case.

**Comment by TurtleOrangina** [comment]
In combination with the ty LSP not showing the unused variable as "grayed out" in vscode this caused a bug for me that slipped through review. I agree that B007 should trigger in the comprehension (or alternatively add unused variable in comprehension as a separate rule).

My example was something like this:

```python
uploaded_image_results = [client.upload("image_path") for image_path in image_paths]
```

The bug was the quotes around image_path as the parameter to client.upload.
